lib/max/Delivery/log.php

```diff
@@ -55,6 +55,18 @@ function MAX_Delivery_log_logAdImpression($adId, $zoneId)
     // Only log impressions if impression logging is enabled
     if (empty($GLOBALS['_MAX']['CONF']['logging']['adImpressions'])) { return true; }
 
+        // If no-impression/click/redirect on inactive banners is enabled,
+        // check to see if the banner is active. If not, exit impression logging
+        // at this point, without recording the impression
+        if (isset($GLOBALS['_MAX']['CONF']['logging']['blockInactiveBanners'])) {
```

Comment:
I think this should be `!empty()`. Which means I missed it during the other review too ;)

How about adding a function `MAX_Delivery_log_isAdBlockedBecauseInactive($adId)` that can be used by both imps and clicks?